### PR TITLE
[NetAppFiles]bump to 2023-07-01

### PR DESCRIPTION
**Pull request by audunn**
## Mandatory Checklist

- Please choose the target release of Azure PowerShell
  - [x] [General release](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Public preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Private preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Engineering build](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] N/A

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For SDK-based development mode, update `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`.
        * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. 
    * For autorest-based development mode, include the changelog in the PR description.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/NetAppFiles/NetAppFiles/ActiveDirectory/RemoveNetAppFilesActiveDirectory.cs [hunk 1]
```diff
@@ -108,8 +109,16 @@ public override void ExecuteCmdlet()
...
-                var updatedAnfAccount = AzureNetAppFilesManagementClient.Accounts.BeginCreateOrUpdate(ResourceGroupName, AccountName, anfAccount);
-                success = true;
+                try
+                {
+                    var updatedAnfAccount = AzureNetAppFilesManagementClient.Accounts.BeginCreateOrUpdate(ResourceGroupName, AccountName, anfAccount);
+                    success = true;
+                }
+                catch (ErrorResponseException ex)
+                {
+                    ex = new ErrorResponseException(ex.Body.Error.Message);
```
Comment (wyunchi-ms) [inline src/NetAppFiles/NetAppFiles/ActiveDirectory/RemoveNetAppFilesActiveDirectory.cs:119]: I suggest to add ex as the inner exception
```suggestion
                    ex = new ErrorResponseException(ex.Body.Error.Message, ex);
```

File: src/NetAppFiles/NetAppFiles/Account/UpdateNetAppFilesAccount.cs [hunk 2]
```diff
@@ -195,8 +195,16 @@ public override void ExecuteCmdlet()
...
-                WriteObject(anfAccount.ConvertToPs());
+                try 
+                { 
+                    var anfAccount = AzureNetAppFilesManagementClient.Accounts.Update(ResourceGroupName, Name, netAppAccountBody);
+                    WriteObject(anfAccount.ConvertToPs());
+                }
+                catch (ErrorResponseException ex)
+                {
+                    ex = new ErrorResponseException(ex.Body.Error.Message);
+                    throw ex;
```
Comment (wyunchi-ms) [inline src/NetAppFiles/NetAppFiles/Account/UpdateNetAppFilesAccount.cs:206]:
```suggestion
                    ex = new ErrorResponseException(ex.Body.Error.Message);
                    throw ex;
```
```suggestion
                    throw new ErrorResponseException(ex.Body.Error.Message, ex);
```

Comment (audunn) [reply src/NetAppFiles/NetAppFiles/Account/UpdateNetAppFilesAccount.cs]: @wyunchi-ms  Updated

**Review by wyunchi-ms — APPROVED** [review]

**Comment by audunn** [comment]
Fixes https://github.com/Azure/azure-powershell/issues/15502

**Comment by audunn** [comment]
Fixes 
https://github.com/Azure/azure-powershell/issues/15502
https://github.com/Azure/azure-powershell/issues/19459

File: src/NetAppFiles/NetAppFiles/Account/GetNetAppFilesAccount.cs [hunk 3]
```diff
@@ -63,21 +64,27 @@ public override void ExecuteCmdlet()
...
+                        var anfAccounts = AzureNetAppFilesManagementClient.Accounts.ListBySubscription().Select(e => e.ConvertToPs());
+                        WriteObject(anfAccounts, true);
+                }
             }
-            else
+            catch (ErrorResponseException ex)
             {
-                var anfAccounts = AzureNetAppFilesManagementClient.Accounts.ListBySubscription().Select(e => e.ConvertToPs());
-                WriteObject(anfAccounts, true);
+                throw new ErrorResponseException(ex.Body.Error.Message, ex);
```
Comment (wyunchi-ms) [inline src/NetAppFiles/NetAppFiles/Account/GetNetAppFilesAccount.cs:87]: We suggest to use `AzPSCloudException`, this exception can be telemtried.

Comment (audunn) [reply src/NetAppFiles/NetAppFiles/Account/GetNetAppFilesAccount.cs:87]: Curiously the Generagted code changed to `ErrorResponseException` when we reference from common types,  updating

**Comment by audunn** [comment]
@wyunchi-ms  updated based on your review

**Review by wyunchi-ms — APPROVED** [review]

**Review by wyunchi-ms — APPROVED** [review]

**Pull request merged**